src/include.am

```diff
@@ -61,9 +61,12 @@ endif
 
 src_libwolfssl_la_SOURCES += \
                wolfcrypt/src/hmac.c \
-               wolfcrypt/src/random.c \
                wolfcrypt/src/hash.c
 
+if BUILD_RNG
+src_libwolfssl_la_SOURCES += wolfcrypt/src/random.c
```

Comment:
You should add #ifndef WC_NO_RNG in random.c as well, for people that are including all .c files.